pkg/resourceinterpreter/configurableinterpreter/luavm/kube.go

```diff
@@ -0,0 +1,93 @@
+package luavm
+
+import (
+	lua "github.com/yuin/gopher-lua"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+const (
+	// KubeLibName is the name of the kube library.
+	KubeLibName = "kube"
+)
+
+// KubeLoader loads the kube library. Import this library before you script by:
+//
+//	local kube = require("kube")
+//
+// Then you can call functions in this library by:
+//
+//	kube.xxx()
+//
+// This library Contains:
+//   - function resourceAdd(r1, r2, ...)
+//     accruing the quantity of resources. Example:
+//     cpu = kube.resourceAdd(r1.cpu, r2.cpu, r3.cpu)
+//   - function accuratePodRequirements(pod) requirements
+//     accurate total resource requirements for pod. Example:
+//     requirements = kube.accuratePodRequirements(pod)
+func KubeLoader(ls *lua.LState) int {
+	mod := ls.SetFuncs(ls.NewTable(), kubeFuncs)
+	ls.Push(mod)
+	return 1
+}
+
+var kubeFuncs = map[string]lua.LGFunction{
+	"resourceAdd":             resourceAdd,
+	"accuratePodRequirements": accuratePodRequirements,
+}
+
+func resourceAdd(ls *lua.LState) int {
+	res := resource.Quantity{}
+	n := ls.GetTop()
+	for i := 1; i <= n; i++ {
+		q := checkResourceQuantity(ls, i)
+		res.Add(q)
+	}
+
+	s := res.String()
+	ls.Push(lua.LString(s))
+	return 1
+}
```

Comment:
When this function would be used?

Reply:
Users may want to accurate resource quantity with their logic, rather `accuratePodRequirements`.